File: algo/111. Minimum Depth of Binary Tree.js
// BFS

/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {
    if (root === null) {
        return 0;
    }
    
    var queue = [root];
    var levelIndex = 1;
    while (queue.length != 0) {
        var levelSize = queue.length;
        for (var i = 0; i < levelSize; i ++) {
            var currNode = queue.shift();
            if (currNode.left === null && currNode.right === null) {
                return levelIndex;
            }
            if (currNode.left) {
                queue.push(currNode.left);
            }
            if (currNode.right) {
                queue.push(currNode.right);
            }
        }
        levelIndex++;
    }
    
};

File: round2/111. Minimum Depth of Binary Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int MinDepth(TreeNode root) {
        if (root == null) return 0;
        int lDepth = MinDepth(root.left);
        int rDepth = MinDepth(root.right);
        if (lDepth == 0) return rDepth + 1;
        if (rDepth == 0) return lDepth + 1;
        return Math.Min(lDepth, rDepth) + 1;
    }
}